test/test_datasets.py

```diff
@@ -1998,5 +1998,78 @@ def test_bad_input(self):
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(
+        split=("train", "test"), pass_name=("clean", "final", "both"), camera=("left", "right", "both")
+    )
+    FEATURE_TYPES = (PIL.Image.Image, PIL.Image.Image, (np.ndarray, type(None)))
+
+    FLOW_H, FLOW_W = 3, 4
+
+    def inject_fake_data(self, tmpdir, config):
+        root = pathlib.Path(tmpdir) / "FlyingThings3D"
+
+        num_images_per_camera = 3 if config["split"] == "train" else 4
+        passes = ("frames_cleanpass", "frames_finalpass")
+        splits = ("TRAIN", "TEST")
+        letters = ("A", "B", "C")
+        subfolders = ("0000", "0001")
+        cameras = ("left", "right")
+        for pass_name in passes:
+            for split in splits:
+                for letter in letters:
+                    for subfolder in subfolders:
+                        current_folder = root / pass_name / split / letter / subfolder
+                        for camera in cameras:
+                            datasets_utils.create_image_folder(
+                                current_folder,
+                                name=camera,
+                                file_name_fn=lambda image_idx: f"00{image_idx}.png",
+                                num_examples=num_images_per_camera,
+                            )
+
+        directions = ("into_future", "into_past")
+        for split in splits:
+            for letter in letters:
+                for subfolder in subfolders:
+                    for direction in directions:
+                        current_folder = root / "optical_flow" / split / letter / subfolder / direction
+                        for camera in cameras:
+                            os.makedirs(str(current_folder / camera))
+                            for i in range(num_images_per_camera):
+                                datasets_utils.make_fake_pfm_file(
+                                    self.FLOW_H, self.FLOW_W, file_name=str(current_folder / camera / f"{i}.pfm")
+                                )
+
+        num_cameras = 2 if config["camera"] == "both" else 1
+        num_passes = 2 if config["pass_name"] == "both" else 1
+        num_examples = (
+            (num_images_per_camera - 1) * num_cameras * len(subfolders) * len(letters) * len(splits) * num_passes
+        )
+        return num_examples
+
+    @datasets_utils.test_all_configs
+    def test_flow(self, config):
+        with self.create_dataset(config=config) as (dataset, _):
+            assert dataset._flow_list and len(dataset._flow_list) == len(dataset._image_list)
+            for _, _, flow in dataset:
+                assert flow.shape == (2, self.FLOW_H, self.FLOW_W)
+                # We don't check the values because the reshaping and flipping makes it hard to figure out
+
+    def test_bad_input(self):
+        with pytest.raises(ValueError, match="split must be either"):
+            with self.create_dataset(split="bad"):
+                pass
```

Comment:
If you go with my suggestion from https://github.com/pytorch/vision/pull/4860#discussion_r743441116, we probably don't need these tests.

Reply:
I'll go with your suggestion but still keep the test 🤓